### PR TITLE
Add authentication

### PR DESCRIPTION
**Pull request by mads256h**
Add endpoints `/accounts/register` & `/accounts/login`

When you register or login successfully you receive an auth token. This needs to be sent in the `X-Auth-Token` header in authenticated endpoints.
To make an endpoint authenticated add the `Authentication` extractor to the endpoint.
This will make the id of the account available to the endpoint. If the authentication token is invalid the request terminates with an error and the endpoint handler is never executed.

### CODE REVIEW

**Comment by mads256h** [comment]
We should maybe use UUID's instead of salt generation for auth tokens

File: src/data_model/mod.rs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 mod device;
 mod event;
+pub mod protocol;
```
Comment (Mikkel-Boje) [inline src/data_model/mod.rs:3]: I do not think protocol is a part of the data model, maybe make a new module

File: src/extractors/auth.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+// Account id
+pub struct Authentication(pub i64);
+
+#[async_trait]
+impl FromRequestParts<SqlitePool> for Authentication {
+    type Rejection = (StatusCode, String);
+
+    async fn from_request_parts(
+        parts: &mut Parts,
```
Comment (Mikkel-Boje) [inline src/extractors/auth.rs:33]: why mut?

**Review by Mikkel-Boje — CHANGES_REQUESTED** [review]
Looks great

File: src/data_model/mod.rs (hunk 1 above)
Comment (mads256h) [reply src/data_model/mod.rs]: Fixed

File: src/extractors/auth.rs (hunk 2 above)
Comment (mads256h) [reply src/extractors/auth.rs:33]: Makes it possible to use in request handlers

**Review by Mikkel-Boje — APPROVED** [review]

**Review by Darkros1245 — APPROVED** [review]

**Pull request merged**